perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1233,6 +1233,32 @@ public RichMember createSponsoredMember(PerunSession session, Vo vo, String name
 		return membersManagerBl.getRichMember(session, membersManagerBl.createSponsoredMember(session, vo, namespace, name, password, sponsor, true));
 	}
 
+	@Override
+	public RichMember setSponsoredMember(PerunSession session, Vo vo, User userToBeSponsored, String namespace, String password, User sponsor)
+		throws PrivilegeException, AlreadyMemberException, LoginNotExistsException, PasswordCreationFailedException,
+		ExtendMembershipException, WrongAttributeValueException, ExtSourceNotExistsException, WrongReferenceAttributeValueException,
+		UserNotInRoleException, PasswordStrengthException, InvalidLoginException {
+		Utils.checkPerunSession(session);
+		Utils.notNull(vo, "vo");
+		Utils.notNull(userToBeSponsored, "userToBeSponsored");
+		Utils.notNull(namespace, "namespace");
+		Utils.notNull(password, "password");
+
+		log.info("setSponsoredMember(vo={},namespace='{}',displayName='{}',sponsor={}", vo.getShortName(), namespace, userToBeSponsored.getFirstName() + " " + userToBeSponsored.getLastName(), sponsor == null ? "null" : sponsor.getId());
+
+		if (sponsor == null) {
+			//sponsor is the caller
+			sponsor = session.getPerunPrincipal().getUser();
+		} else {
+			//Authorization
+			if (!AuthzResolver.authorizedInternal(session, "setSponsoredMember_Vo_User_String_String_User_policy", Arrays.asList(vo, sponsor))) {
```

Comment:
```suggestion
			if (!AuthzResolver.authorizedInternal(session, "setSponsoredMember_Vo_User_String_String_User_policy", vo, sponsor)) {
```
We have made some changes recently, and now this can be simplified.